lib/diagnostic.rb

```diff
@@ -62,8 +80,13 @@ def initialize
 
 ##
 # your answers here
-Answer.housecat_noise = nil
-#
+Answer.housecat_noise = "I am a Animal, and I go 'meow'"
+# Housecat.new creates a new instance of housecat, which inherits from cat. Cat
+# inherits from Animal. Calling the "say_hello" method would look first to the
+# housecat class and then to the cat class, unsuccessfully trying to find that
+# method. Finally it would look to the Animal class, where the "say_hello" is
+# written, and return the associated string. Because @sound is an instance
+# variable of the Housecat class, it is not overwritten by nil.
```

Comment:
Your reasoning is good, but you're mistaken about how the method call ultimately works. 
Although the method was written on the Animal class, self's value is determined at _runtime_. Since the method was called on `HouseCat.new`, at runtime the instance we were working with was a HouseCat, so `self` evaluates to `HouseCat`. It's almost as if the `say_hello` method were written within the HouseCat class.